src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -502,8 +457,71 @@ private static bool LoadNetFrameworkDelegates(string userSpecifiedPath)
 
             return SetDelegates(typeof(NativeMethods.NativeMethods));
         }
+
 #endif
 
+        private static bool TrySetDelegates(List<Type> nativeMethodCandidateTypes)
+        {
+            foreach (var t in nativeMethodCandidateTypes)
+            {
+                if (SetDelegates(t))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private static bool LoadNetStandardDelegates(string userSpecifiedPath)
+        {
+            if (userSpecifiedPath != null)
+            {
+                if (WindowsNative.LoadLibraryEx(userSpecifiedPath, IntPtr.Zero, WindowsNative.LoadLibraryFlags.LOAD_WITH_ALTERED_SEARCH_PATH) == IntPtr.Zero)
+                {
+                    // TODO: The Win32Exception class is not available in .NET Standard, which is the easy way to get the message string corresponding to
+                    // a win32 error. FormatMessage is not straightforward to p/invoke, so leaving this as a job for another day.
+                    throw new InvalidOperationException($"Failed to load librdkafka at location '{userSpecifiedPath}'. Win32 error: {Marshal.GetLastWin32Error()}");
+                }
+            }
+            return TrySetDelegates(new List<Type> { typeof(NativeMethods.NativeMethods) });
+        }
+
+        private static bool LoadOSXDelegates(string userSpecifiedPath)
+        {
+            if (userSpecifiedPath != null)
+            {
+                if (PosixNative.dlopen(userSpecifiedPath, RTLD_NOW) == IntPtr.Zero)
+                {
+                    throw new InvalidOperationException($"Failed to load librdkafka at location '{userSpecifiedPath}'. dlerror: '{PosixNative.LastError}'.");
+                }
+            }
+
+            return TrySetDelegates(new List<Type> { typeof(NativeMethods.NativeMethods) });
+        }
+
+        private static bool LoadLinuxDelegates(string userSpecifiedPath)
```

Comment:
Since LoadOSX and LinuxDelegates are identical (from what I can see), and it is really a POSIX API, maybe call it LoadPosixDelegates()?

Reply:
the LinuxDelegates method tries up to 4 different `NativeMethods` classes to enable the alpine, centos and debian .so's, so it's pretty tied to linux. the macos one just tries the general one.